internal/v4/search.go

```
@@ -0,0 +1,83 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package v4
+
+import (
+	//	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/juju/errgo"
+
+	"github.com/juju/charmstore/internal/charmstore"
+	"github.com/juju/charmstore/internal/router"
+	"github.com/juju/charmstore/params"
+)
+
+// GET search[?text=text][&autocomplete=1][&filter=value…][&limit=limit][&include=meta]
+// http://tinyurl.com/qzobc69
+func (h *Handler) serveSearch(w http.ResponseWriter, req *http.Request) {
```

If you change this to:

```
func (h *Handler) serveSearch(w http.ResponseWriter, req *http.Request) (interface{}, error) {
```

then we can wrap it using HandleJSON and you can just return the errors
and the final result. It would also be consistent with the other handlers.